### ISSUE TITLE
Incrementor in for loop not corrected

### ISSUE THREAD

**Issue by breerly** [FriendsOfPHP/PHP-CS-Fixer#81]
Incrementor in for loop is not being corrected: 

```
for ($i = 0; $i < $length; $i ++){
```

is not converted to the expected

```
for ($i = 0; $i < $length; $i++){
```

In other words, `$i ++` should be converted to `$i++`

**Comment by vbence** [comment]
It does not say, just gives an example, which is quite insufficient knowing the various uses of the C-style for loop.

What I get away from this:

 * Operators with two operands (= < > etc.) should be surrounded by spaces (I wonder if this is not covered elsewhere).
 * There should be no space between the variable and the `++`.

---

Excerpt from [PSR-2][1]:

### 5.4. `for`

A `for` statement looks like the following. Note the placement of parentheses,
spaces, and braces.

```php
<?php
for ($i = 0; $i < 10; $i++) {
    // for body
}
```

[1]: https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md

**Comment by keradus** [comment]
Operators like `<` aren't relevant here.
Example didn't say that one should use post- or pre- incrementation.

**Comment by vbence** [comment]
I think the whole spacing issue for loop deserves to be looked at.  Whether or not pre-incrementation is allowed I think the following examples are all wrong:
```
// too many spaces
for ($i = 0; $i < 10; $i ++) {

// too many spaces
for ($i = 0; $i < 10; ++ $i) {

// too few spaces
for ($i=0; $i<10; $i++) {

// incredibly few spaces
for ($i=0;$i<10;$i++) {
```

**Comment by stof** [comment]
@vbence most of the cases in your comment are already fixed by the fixer enforcing spaces around binary operators

**Comment by vbence** [comment]
@stof I think **operators_spaces [symfony]** is the one, but it is outside PSR-2 level.

**Comment by stof** [comment]
@vbence indeed, because PSR-2 does not define any rules about using spaces or no around operators.

**Comment by vbence** [comment]
@stof I guess it's up for interpretation. I quoted the doc in my first comment.

Are you saying the following is valid PSR2?
```php
   for ($i=0;$i<10;$i++) {
```

**Comment by stof** [comment]
@vbence read what the section 5 is defining in the spec. Your snippet is technically valid, because it has the parentheses, braces and spaces in the places defined in the spec. The section 5 is *not* about spaces around operators or between statements (Symfony on the other hand defines such a rule, making your snippet invalid for the Symfony coding standards)

**Comment by gharlan** [comment]
This issue is fixed by #1064 and can be closed.

**Comment by keradus** [comment]
Thank you. Closing
